pkg/wallet/account/account.go

```diff
@@ -0,0 +1,246 @@
+package account
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"errors"
+	"fmt"
+
+	"github.com/awnumar/memguard"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/station-massa-wallet/pkg/crypto"
+	"github.com/massalabs/station-massa-wallet/pkg/types"
+	"github.com/massalabs/station-massa-wallet/pkg/types/object"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	AccountLastVersion    = 1
+	AccountUnknownVersion = 0
+)
+
+var (
+	ErrInvalidPrivateKey = errors.New("invalid private key")
+	ErrGeneratingKeyPair = errors.New("generating ed25519 key-pair")
+	ErrInvalidParameter  = errors.New("invalid parameter")
+)
+
+type Account struct {
+	Version      uint8                      `yaml:"Version"`
+	Nickname     string                     `yaml:"Nickname,omitempty"`
+	Address      *types.Address             `yaml:"Address,omitempty"`
```

Comment:
Why do youo have omitempty here and not elsewhere ?

Reply:
nickname and address are optional fields